parser-typechecker/src/Unison/Codebase/SqliteCodebase/Branch/Cache.hs

```diff
@@ -0,0 +1,49 @@
+module Unison.Codebase.SqliteCodebase.Branch.Cache where
+
+import qualified Data.Map as Map
+import System.Mem.Weak
+import qualified U.Codebase.HashTags as V2
+import qualified Unison.Codebase.Branch as V1.Branch
+import Unison.Prelude
+import qualified Unison.Sqlite as Sqlite
+import UnliftIO.STM
+
+-- | A cache of 'V1.Branch.Branch' by 'V2.CausalHash'es.
+data BranchCache m = BranchCache
```

Comment:
I had to use a record here rather than individual functions since the place where it's used is fully polymorphic over `m`, but is eventually run inside a `Sqlite.Transaction`, so I can't add a `MonadIO` constraint, nor can I use `Sqlite.unsafeIO` directly.

Maybe we should consider monomorphizing some of the operations in `Unison.Codebase.SqliteCodebase.Conversions` to Transaction?

This works, but is probably more confusing than necessary.

Reply:
I think this is fine; it could also just have functions in `IO`, and we could use `Sqlite.unsafeIO` at the call sites in `getRootBranch` / `getBranchForHash`